docs/content/labs/IPv4/dns.md

```diff
@@ -0,0 +1,105 @@
+dhcp-option=ipv4,option:router,192.168.125.1
+
+resolv-file=/opt/dnsmasq/upstream-resolv.conf
+except-interface=lo
+dhcp-lease-max=81
+log-dhcp
+no-hosts
+
+# DHCP Reservations
+dhcp-leasefile=/opt/dnsmasq/hosts.leases
+
+# Include all files in a directory depending on the suffix
+conf-dir=/opt/dnsmasq/include.d/*.ipv4
+```
+
+Create the upstream resolver to delegate the non-local environments queries
+
+- `/opt/dnsmasq/upstream-resolv.conf`
+```
+nameserver 8.8.8.8
+nameserver 8.8.4.4
+```
+
+Create the different component DNS configurations
+
+- `/opt/dnsmasq/include.d/hosted-nodeport.ipv4`
+```
+## Nodeport
+host-record=api-int.hosted-ipv4.hypershiftbm.lab,192.168.125.20
+host-record=api-int.hosted-ipv4.hypershiftbm.lab,192.168.125.21
+host-record=api-int.hosted-ipv4.hypershiftbm.lab,192.168.125.22
+host-record=api.hosted-ipv4.hypershiftbm.lab,192.168.125.20
+host-record=api.hosted-ipv4.hypershiftbm.lab,192.168.125.21
+host-record=api.hosted-ipv4.hypershiftbm.lab,192.168.125.22
+
+## Nodeport
+## IMPORTANT!: You should point to the node which is exposing the router.
+## You can also use MetalLB to expose the Apps wildcard.
+address=/apps.hosted-ipv4.hypershiftbm.lab/192.168.125.30
+
+## General
+dhcp-host=aa:aa:aa:aa:02:01,hosted-ipv4-worker0,192.168.125.30
+dhcp-host=aa:aa:aa:aa:02:02,hosted-ipv4-worker1,192.168.125.31
+dhcp-host=aa:aa:aa:aa:02:03,hosted-ipv4-worker2,192.168.125.32
+```
+
+- `/opt/dnsmasq/include.d/hub.ipv4`
+```
+host-record=api-int.hub-ipv4.hypershiftbm.lab,192.168.125.10
+host-record=api.hub-ipv4.hypershiftbm.lab,192.168.125.10
+address=/apps.hub-ipv4.hypershiftbm.lab/192.168.125.11
+dhcp-host=aa:aa:aa:aa:02:01,ocp-master-0,192.168.125.20
+dhcp-host=aa:aa:aa:aa:02:02,ocp-master-1,192.168.125.21
+dhcp-host=aa:aa:aa:aa:02:03,ocp-master-2,192.168.125.22
+dhcp-host=aa:aa:aa:aa:02:06,ocp-installer,192.168.125.25
+dhcp-host=aa:aa:aa:aa:02:10,ocp-bootstrap,192.168.125.26
+```
+
+- `/opt/dnsmasq/include.d/infra.ipv4`
+```
+host-record=registry.hypershiftbm.lab,192.168.125.1
+```
+
+To proceed, we must create a systemd service for the management of the dnsmasq service and disable the system's default dnsmasq service:
+
+- `/etc/systemd/system/dnsmasq-virt.service`
+```
+[Unit]
+Description=DNS server for Openshift 4 Clusters.
+After=network.target
+
+[Service]
+User=root
+Group=root
+ExecStart=/usr/sbin/dnsmasq -k --conf-file=/opt/dnsmasq/dnsmasq.conf
+
+[Install]
+WantedBy=multi-user.target
+```
+
+The commands to do so:
+
+```
+systemctl daemon-reload
+systemctl disable --now dnsmasq
+systemctl enable --now dnsmasq-virt
+```
+
+!!! note
+
+    This step is mandatory for both Disconnected and Connected environments. Additionally, it holds significance for both Virtualized and Bare Metal environments. The key distinction lies in the location where the resources will be configured. In a non-virtualized environment, a more robust solution like Bind is recommended instead of a lightweight dnsmasq.
```

Comment:
@jparrill If this step is mandatory for both connected and disconnected environments, does it override the info that we already have published for configuring DNS in the official docs? 
You can see the source for the official docs related to DNS here: https://github.com/stolostron/rhacm-docs/blob/2.9_stage/clusters/hosted_control_planes/hosted_bare_metal_dns.adoc and https://github.com/stolostron/rhacm-docs/blob/2.9_stage/clusters/hosted_control_planes/hosted_bare_metal_dns.adoc.

Reply:
Good point, I don't think so (actually it's mostly the same), the only thing I would add is the same but in IPv6 and Dual stack (I made up the IP addresses).

which in IPv6 would be:

```
api.example.krnl.es.    IN A 2620:52:0:1306::5
api.example.krnl.es.    IN A 2620:52:0:1306::6
api.example.krnl.es.    IN A 2620:52:0:1306::7
api-int.example.krnl.es.    IN A 2620:52:0:1306::5
api-int.example.krnl.es.    IN A 2620:52:0:1306::6
api-int.example.krnl.es.    IN A 2620:52:0:1306::7
`*`.apps.example.krnl.es. IN A 2620:52:0:1306::10
```

And for Dual stack, they should resolve in both stacks depending how do you execute the dig command `dig A google.com` (For IPv4) or `dig AAAA google.com` (For IPv6)

Reply:
@jparrill When I move this content to the official docs, do you think it would be best if I mentioned the already existing docs about DNS as a prereq to this? I'm trying to figure out what the best way is to present this info keeping in mind that we have other docs about DNS as well.

Reply:
Reuse existing documentation will be always better (IMHO), only makes sense to create new doc if the case differs from the current one.